File: js/geoip.mjs
import request from '@surfy/request';

/*

Geo IP

*/

class GeoIP {

	constructor(conf){
		this.email = conf.email;
		this.key = conf.key;
		this.headers = {
			email: this.email,
			key: this.key
		};
	}

	async get(IP){
		let result = await request({
			url: `https://api.surfy.one/geo/ip/${IP}`,
			method: 'POST',
			headers: this.headers
		});

		return result;
	}
};

export default GeoIP;

File: py/geoip.py
import requests

class GeoIP():

	def __init__(self, email, key):
		self.email = email
		self.key = key
		self.headers = { 'Email' : self.email, 'Key' : self.key }

	def get(self, IP):
		url = 'https://api.surfy.one/geo/ip/' + IP
		result = requests.post(url, headers = self.headers)
		return result.json()